src/integration_test/java/com/klodnicki/Bike/v2/rest/controller/RentBikeControllerIntegrationTest.java

```diff
@@ -0,0 +1,226 @@
+
+        RentResponseDTO expected = new RentResponseDTO(1L, rentRequestDTO.getRentalStartTime(),
+                rentRequestDTO.getRentalEndTime(), rentRequestDTO.getDaysOfRent(), rentRequestDTO.getBikeForNormalUserResponseDTO(),
+                rentRequestDTO.getUserForNormalUserResponseDTO(), rentRequestDTO.getStationForNormalUserResponseDTO());
+
+        webTestClient.put()
+                .uri("/api/rentals/" + rent.getId())
+                .bodyValue(rentRequestDTO)
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(RentResponseDTO.class)
+                .consumeWith(response -> {
+                    RentResponseDTO responseDTO = response.getResponseBody();
+                    assertNotNull(responseDTO);
+                    assertEquals(expected.getDaysOfRent(), responseDTO.getDaysOfRent());
+                });
+
+        rent.setBike(null);
+        rent.setUser(null);
+        rentRepository.save(rent);
+    }
+
+    @Test
+    public void returnBike_ShouldReturnResponseEntityAsStringAndReturnBikeToStation_WhenRentAndStationIdsAreGiven() {
+
+        Rent rent = new Rent(null, LocalDateTime.of(2023, 11, 23, 10, 0, 0),
+                null, 10, 100.00, bike, user, null);
+        rentRepository.save(rent);
+
+        webTestClient.put()
+                .uri(uriBuilder -> uriBuilder
+                        .path("/api/returns/" + rent.getId())
+                        .queryParam("returnChargingStationId", chargingStation.getId())
+                        .build())
+                .exchange()
+                .expectStatus().isOk()
+                .expectBody(String.class)
+                .consumeWith(response -> {
+                    String responseMessage = response.getResponseBody();
+                    ChargingStation returnChargingStation =
+                            chargingStationRepository.findById(chargingStation.getId()).orElseThrow(IllegalArgumentException::new);
+                    assertNotNull(responseMessage);
+                    assertEquals("Bike successfully returned.", responseMessage);
+                    assertEquals(chargingStation.getFreeSlots() - 1, returnChargingStation.getFreeSlots());
+                    assertEquals(rent.getAmountToBePaid(), returnChargingStation.getRent().getAmountToBePaid());
+                    assertEquals(rent.getUser().getBalance() - rent.getAmountToBePaid(), returnChargingStation.getRent().getUser().getBalance());
+                    assertFalse(returnChargingStation.getBikeList().isEmpty());
+                    //I use stream in assertion to check if on the bikeList is the bike with the specific id
+                    assertTrue(returnChargingStation.getBikeList().stream().anyMatch(b -> b.getId().equals(bike.getId())));
+                });
+        rent.setBike(null);
+        rent.setUser(null);
+        bike.setChargingStation(null);
+        rentRepository.save(rent);
+        bikeRepository.save(bike);
```

Comment:
Again, what does lines here are doing?

Reply:
If I dont add it then Im not able to delete these on db (they hold foreing keys and cascades dont work)